core/src/main/scala/shapeless/ops/coproduct.scala

```diff
@@ -859,6 +859,7 @@ object coproduct {
 
     def head(c: C): Option[H]
     def tail(c: C): Option[T]
+    def cons(e: Either[H, T]): H :+: T
```

Comment:
Close ... this should give us a `C` in the same way that `cons` on `IsHCons` gives us an `L`.